fsm.go

```diff
@@ -108,27 +95,19 @@ func (fsm *FSM) Snapshot() (raft.FSMSnapshot, error) {
 		return nil, errors.New("cannot snapshot inconsistent state")
 	}
 
-	// Encode the state
-	bytes, err := EncodeSnapshot(fsm.stateWrap.State)
-	if err != nil {
-		return nil, err
-	}
-
-	var snap fsmSnapshot = bytes
-	return snap, nil
+	return &fsmSnapshot{
```

Comment:
Could we one line this struct so the second return value is more obvious please?